pkg/registry/extensions/deployment/strategy_test.go

```diff
@@ -78,3 +80,70 @@ func newDeployment(labels, annotations map[string]string) *extensions.Deployment
 		},
 	}
 }
+
+func TestSelectorImmutability(t *testing.T) {
+	tests := []struct {
+		url                    string
+		oldSelectorLabels      map[string]string
+		newSelectorLabels      map[string]string
+		expectedAPIGroup       string
+		expectedAPIVersion     string
+		expectedSelectorLabels map[string]string
+	}{
+		{"/apis/apps/v1beta2/namespaces/default/deployment/test-deployment", map[string]string{"a": "b"}, map[string]string{"c": "d"}, "apps", "v1beta2", map[string]string{"a": "b"}},
```

Comment:
I just remembered, Deployment already existed in `apps/v1beta1`, so we should add a test case to verify that remains mutable.